### PR TITLE
Implement $?

### PR DESCRIPTION
**Pull request by takumasa0222**
Modified expand code to apply 0 and modified execute_logical_operator to set proper is_exec_child_flag and  add some test code to expand_test.sh. Almost all of the test has passed.

Related Issue No:
#48 

## Release Notes

- **New Features**
  - Added support for special variable `?` to retrieve last command exit status.
  - Enhanced visual presentation of test summaries with stylized ASCII art headers.

- **Bug Fixes**
  - Improved error handling in variable expansion and command execution.
  - Fixed potential issues with environment variable retrieval.

- **Tests**
  - Added comprehensive test cases for shell command scenarios.
  - Expanded test script functionality with new assertions focused on exit statuses and edge cases.

- **Chores**
  - Updated test infrastructure and script configurations.
  - Minor code cleanup and documentation improvements.

### CODE REVIEW

File: combination_test/run_test.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+█!/bin/bash
```
[the remark this answers is not in the document]
Comment (takumasa0222) [reply combination_test/run_test.sh]: fixed
